File: gestionart_backend_ruben_y_jessica/api/src/main/java/com/gestionart/api/domain/models/LineaPedido.java
package com.gestionart.api.domain.models;

public class LineaPedido {
    private long id;
    private long idPedido;
    private long idArticulo;
    private int cantidad;
    private double precioUnitario;
   
    public LineaPedido() {
    }

    public LineaPedido(long id,long idPedido, long idArticulo, int cantidad, double precioUnitario) {
        this.id = id;
        this.idPedido = idPedido;
        this.idArticulo = idArticulo;
        this.cantidad = cantidad;
        this.precioUnitario = precioUnitario;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getIdArticulo() {
        return idArticulo;
    }

    public void setIdArticulo(long idArticulo) {
        this.idArticulo = idArticulo;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public double getPrecioUnitario() {
        return precioUnitario;
    }

    public void setPrecioUnitario(double precioUnitario) {
        this.precioUnitario = precioUnitario;
    }

    public long getIdPedido() {
        return idPedido;
    }

    public void setIdPedido(long idPedido) {
        this.idPedido = idPedido;
    }

}

File: gestionart_frontend_ruben_y_jessica/lib/data/models/LineaPedido.dart


class Lineapedido {
  final int id;
  final int idPedido;
  final int idArticulo;
  final int cantidad;
  final double precioUnitario;

  Lineapedido({required this.id, required this.idPedido, required this.idArticulo, required this.cantidad, required this.precioUnitario});

  int getId() {
    return id;
  }
  int getIdPedido() {
    return idPedido;
  }
  int getIdArticulo() {
    return idArticulo;
  }
  int getCantidad() {
    return cantidad;
  }
  double getPrecioUnitario() {
    return precioUnitario;
  }

  factory Lineapedido.fromJson(Map<String, dynamic> json) {
    return Lineapedido(
      id: json['id'] ?? 0,
      idPedido: json['idPedido'] ?? 0,
      idArticulo: json['idArticulo'] ?? 0,
      cantidad: json['cantidad'] ?? 0,
      precioUnitario: (double.tryParse(json['precioUnitario'])) ?? 0.0
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id ?? 0,
      'idPedido': idPedido ?? 0,
      'idArticulo': idArticulo ?? 0,
      'cantidad': cantidad ?? 0,
      'precioUnitario': precioUnitario ?? 0.0
    };
  }
  
}